libraries/aws/provisioning/test/system/aws_iot_tests_provisioning_system.c

```diff
@@ -227,8 +282,61 @@ static void _storeCertificateDataForProvisioning( void * contextParam,
     }
     else
     {
-        IotLogInfo( "Request for new credentials was rejected! Test will be aborted." );
-        AwsIotProvisioning_Assert( false );
+        TEST_FAIL_MESSAGE( "Request for new credentials was rejected! Aborting Test." );
+    }
+}
+
+/*-----------------------------------------------------------*/
+
+static void _storeDataOfCerFromCsrResponseCallback( void * contextParam,
+                                                    const AwsIotProvisioningCreateCertFromCsrResponse_t * pResponseInfo )
+{
+    _certIdAndOwnershipTokenContext_t * contextData =
+        ( _certIdAndOwnershipTokenContext_t * ) contextParam;
+
+    IotLogInfo( "Received StatusCode={%d}", pResponseInfo->statusCode );
+
+    if( pResponseInfo->statusCode == AWS_IOT_PROVISIONING_SERVER_STATUS_ACCEPTED )
+    {
+        /* Allocate buffer space for storing the certificate ID obtained from the server. */
+        contextData->pCertificateIdBuffer =
+            IotTest_Malloc( pResponseInfo->u.acceptedResponse.certIdLength + 1 );
```

Comment:
I always test asserted that what was returned from malloc was not null to be safe in all my tests. I saw this was not done in the BLE tests, but it was done in MQTT tests. I think we should do this here as well.

Reply:
Added null-check asserts for all malloc calls